pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -2102,6 +2103,11 @@ func reportGatewayStatus(
 		if len(addressesToReport) > 0 {
 			gs.Addresses = make([]k8sbeta.GatewayStatusAddress, 0, len(addressesToReport))
 			for _, addr := range addressesToReport {
+				if _, err := netip.ParseAddr(addr); err == nil {
+					addrType = k8s.IPAddressType
```

Comment:
nit: we probably don't to set addrType above if we are always overwriting it

Reply:
it's being used in the `if len(addressesToReport) == 0` above, right?